loky/backend/utils.py

```diff
@@ -84,3 +94,207 @@ def _recursive_terminate(pid):
+                dynamic_threadpool_size[clib] = True
+            except NotImplementedError:
+                dynamic_threadpool_size[clib] = False
+        return dynamic_threadpool_size
+
+    def get_thread_limits(self):
+        limits = {}
+        for clib in self.SUPPORTED_CLIB:
+            try:
+                _get = getattr(self, "{}_get_max_threads".format(clib))
+                limits[clib] = _get()
+            except NotImplementedError:
+                limits[clib] = None
+        return limits
+
+    def _load_openblas(self):
+        self.lib_openblas = self._load_lib(OPENBLAS_MODULE_NAME,
+                                           "openblas_info")
+
+    def openblas_set_num_threads(self, num_threads):
+        if self.lib_openblas is None:
+            raise NotImplementedError("Could not find OpenBLAS library.")
+        self.lib_openblas.openblas_set_num_threads(num_threads)
+
+    def openblas_get_max_threads(self):
+        if self.lib_openblas is None:
+            raise NotImplementedError("Could not find OpenBLAS library.")
+        return self.lib_openblas.openblas_get_num_threads()
+
+    def _load_omp(self):
+        self.lib_gomp = self._load_lib(OMP_MODULE_NAME_GNU, "")
+        self.lib_iomp = self._load_lib(OMP_MODULE_NAME_INTEL, "")
+
+    def openmp_gnu_set_num_threads(self, num_threads):
+
+        if self.lib_gomp is None:
+            raise NotImplementedError("Could not find OpenMP library")
+        self.lib_gomp.omp_set_num_threads(num_threads)
+
+    def openmp_gnu_get_max_threads(self):
+        if self.lib_gomp is None:
+            raise NotImplementedError("Could not find OpenMP library")
+        return self.lib_gomp.omp_get_max_threads()
+
+    def openmp_intel_set_num_threads(self, num_threads):
+        if self.lib_iomp is None:
+            raise NotImplementedError("Could not find OpenMP library")
+        self.lib_iomp.omp_set_num_threads(num_threads)
+
+    def openmp_intel_get_max_threads(self):
+        if self.lib_iomp is None:
+            raise NotImplementedError("Could not find OpenMP library")
+        return self.lib_iomp.omp_get_max_threads()
+
+    def _load_mkl(self):
+        self.lib_mkl = self._load_lib(MKL_MODULE_NAME, "blas_mkl_info")
+
+    def mkl_set_num_threads(self, num_threads):
+        if self.lib_mkl is None:
+            raise NotImplementedError("Could not find MKL libray")
+        self.lib_mkl.MKL_Set_Num_Threads(num_threads)
+
+    def mkl_get_max_threads(self):
+        if self.lib_mkl is None:
+            raise NotImplementedError("Could not find MKL libray")
+        return self.lib_mkl.MKL_Get_Max_Threads()
+
+    def _load_lib(self, module_name, lib_info):
+        lib_name = find_library(module_name)
+        if lib_name is not None:
+            return ctypes.CDLL(lib_name, use_errno=True)
+        return self._get_lib_from_dynamic_libs(module_name)
+
+    def _get_lib_from_dynamic_libs(self, module_name):
+
+        global _MODULE_NAME
```

Comment:
This should be a thread local, otherwise `_get_lib_from_dynamic_libs` is not thread-safe.